modules/programs/taskwarrior.nix

```
@@ -99,5 +101,27 @@ in {
 
       ${cfg.extraConfig}
     '';
+
+    home.activation.regenDotTaskRc = hm.dag.entryAfter [ "writeBoundary" ] ''
+      $VERBOSE_ECHO "Ensuring generated taskwarrior config included in taskrc"
+
+      # Ensure file's existence
+      if [[ ! -s "${userConf}" ]]; then
+        if [[ -v DRY_RUN ]]; then
+          $DRY_RUN_CMD echo "include ${homeConf}" ">" "${userConf}"
+        else
+          echo "include ${homeConf}" > "${userConf}"
+        fi
+      fi
+
+      # Add include statement for home-manager generated config
+      if ! ${pkgs.gnugrep}/bin/grep -qF "include ${homeConf}" ${userConf}; then
+        if [[ -v DRY_RUN ]]; then
+          $DRY_RUN_CMD ${pkgs.gnused}/bin/sed -i '1i include ${homeConf}' ${userConf}
+        else
+          ${pkgs.gnused}/bin/sed -i '1i include ${homeConf}' ${userConf}
+        fi
+      fi
```

nit: this should be an `elif` because the two branches are mutually exclusive and the `DRY_RUN` check is unnecessary in this case.

```suggestion
      elif ! grep -qF "include ${homeConf}" ${userConf}; then
        $DRY_RUN_CMD ${pkgs.gnused}/bin/sed -i '1i include ${homeConf}' ${userConf}
      fi
```

---

I tried applying this patch manually/locally, and I just don't get it, sorry =/

Mechanically deleting lines 115-124, and inserting your 115-117, doesn't give me "compile-able" bash, so the below is my guess based on your comment:

<details>
<summary>New version</summary>

```bash
$VERBOSE_ECHO "Ensuring generated taskwarrior config included in taskrc"

# Ensure file's existence
if [[ ! -s "${userConf}" ]]; then
  if [[ -v DRY_RUN ]]; then
    $DRY_RUN_CMD echo "include ${homeConf}" ">" "${userConf}"
  else
    echo "include ${homeConf}" > "${userConf}"
  fi
fi

# Add include statement for home-manager generated config
if ! grep -qF "include ${homeConf}" ${userConf}; then
  $DRY_RUN_CMD ${pkgs.gnused}/bin/sed -i '1i include ${homeConf}' ${userConf}
fi
```

</details>
---

I struggle with groking the rationale for your suggested change, and will try to write up my understanding next.
___But before you spend time/effort reading and groking it, I just wanna make clear that I don't want to oppose the change if you're certain___ =)

**My Rationale**
1. The `~/.config/task/taskrc` file is never guaranteed to exist, be readable, or empty between `switch` invocations.
    - Hence, the bash `-s` operator reemed fetting to ensure the existance of a non-empty file.
1.  Since the file might exist _and_ be non-empty, the next factor to consider is the preservation of manual edits to the file by the user (like the stateful `context=` variable the originating issue for this PR mentions).
    - Thus, I don't get why the `sed` commands that inserts the HM `include` line is prefixed with `$DRY_RUN_CMD` - this should always run when the preceding grep fails, no? "Independently" of whether it's to be actually invoked, or if its just to be mentioned to the invoking user in a dry-run.


Maybe this is down to my lack of understanding of how `DRY_RUN(_CMD)` works...

---

> Maybe this is down to my lack of understanding of how `DRY_RUN(_CMD)` works...

Yup, that it was:
```bash
if [[ -v DRY_RUN ]] ; then
    _i "This is a dry run"
    export DRY_RUN_CMD=echo
else
    $VERBOSE_RUN _i "This is a live run"
    export DRY_RUN_CMD=""
fi
```

I could not find any documentation mentioning this, maybe it's an idea to "smooth" this "speedbump" for other newcoming developers?

Adding diff in previous post manually.

---

Oh, I see you already did in commit https://github.com/nix-community/home-manager/pull/2761/commits/279ddae8757f7e47cc332f4f47d2045192f5b292 @berbiche! =)